ingestion_tools/scripts/common/image.py

```diff
@@ -375,12 +386,45 @@ def update_headers(self, mrcfile: MrcFile, header_mapper, voxel_spacing):
 
 
 class MaskConverter(TomoConverter):
-    def __init__(self, fs: FileSystemApi, filename: str, label: int = 1, header_only: bool = False):
-        super().__init__(fs, filename, header_only)
+    def __init__(
+        self,
+        fs: FileSystemApi,
+        filename: str,
+        label: int = 1,
+        header_only: bool = False,
+        scale_0_dims: tuple[int, int, int] | None = None,
+        threshold: float | None = None,
+    ):
+        super().__init__(fs=fs, filename=filename, header_only=header_only, scale_0_dims=scale_0_dims)
         self.label = label
+        self.threshold = threshold
 
     def get_pyramid_base_data(self) -> np.ndarray:
-        return (self.volume_reader.get_pyramid_base_data() == self.label).astype(np.int8)
+        # When thresholding, do it after scaling. When already binary, extract the mask before scaling.
+        if self.threshold is not None and self.scale_0_dims is not None:
+            scale_before = True
+            scale_after = False
+        elif self.threshold is None and self.scale_0_dims is not None:
+            scale_before = False
+            scale_after = True
+        else:
+            scale_before = False
+            scale_after = False
+
+        if scale_before:
+            data = resize_local_mean(self.volume_reader.get_pyramid_base_data(), self.scale_0_dims, preserve_range=True)
+        else:
+            data = self.volume_reader.get_pyramid_base_data()
+
+        if self.threshold is not None:
+            data = (data >= self.threshold).astype(np.int8)
+
+        data = (data == self.label).astype(np.int8)
```

Comment:
Should this check only be done when the threshold value is `None`?

Reply:
Yes, the idea is that if threshold is not set, it is not applied.

Reply:
Now only comparing against the label when not applying a threshold.